src/formatting_stack/util/diff.clj

```diff
@@ -0,0 +1,46 @@
+(ns formatting-stack.util.diff
+  (:require
+   [clojure.java.data :refer [from-java]]
+   [clojure.java.io :as io]
+   [clojure.spec.alpha :as spec]
+   [clojure.string :as string]
+   [formatting-stack.util :refer [rcomp]]
+   [nedap.speced.def :as speced]
+   [nedap.utils.spec.predicates :refer [present-string?]])
+  (:import
+   (io.reflectoring.diffparser.api UnifiedDiffParser)))
+
+(spec/def ::begin pos-int?)
+(spec/def ::end pos-int?)
+(spec/def ::filename present-string?)
+
+(spec/def ::line-numbers
+  (spec/coll-of (spec/keys :req-un [::begin ::end ::filename])))
+
+(speced/defn ^::line-numbers diff->line-numbers
+  "Returns maps of consecutive removals/changes (as `:begin` to `:end`) per `:filename` in `diff`."
+  [^string? diff]
+  (->> (io/input-stream (.getBytes diff))
+       (.parse (UnifiedDiffParser.))
```

Comment:
Please don't use ->> for java interop

Only ->

This style is 100% consistent in formatting-stack, speced.def et al

The intent being to standardize what these can be used for, minimizing mental overhead (i.e. I shouldn't have to thread in my head)